### ISSUE TITLE
Alerting: yaml for alerts that is exported through web UI is causing grafana to crash on startup

### ISSUE THREAD

**Issue by danfinn** [grafana/grafana#97804]
### What happened?

grafana 11.3.0 on k8s

Our general workflow is that we create alerts via the UI, we then export the yaml for those alerts from the UI and save that into git.  When grafana is restarted it pulls in this file for provisioning from git.

This morning I added some new alerts and I noticed we are getting a strange error when the grafana pod attempts to restart:
```
logger=ngalert.notifier.alertmanager org=1 t=2024-12-11T16:20:05.501464632Z level=info msg="Applying new configuration to Alertmanager" configHash=daf3073212d3795a32b739cee60d6728
logger=ngalert.state.manager t=2024-12-11T16:20:05.512927075Z level=info msg="Running in alternative execution of Error/NoData mode"
logger=infra.usagestats.collector t=2024-12-11T16:20:05.514848677Z level=info msg="registering usage stat providers" usageStatsProvidersLen=2
logger=provisioning t=2024-12-11T16:20:06.709714942Z level=error msg="Failed to provision alerting" error="failure to map file alerts.yaml: failure parsing rules: rule 'New Blocked SQLs (4 hour interval)' failed to parse: empty duration string"
Error: ✗ failure to map file alerts.yaml: failure parsing rules: rule 'New Blocked SQLs (4 hour interval)' failed to parse: empty duration string
```

so what I've done to try and resolve this is to add the following line to the alerts:

`for: 0s`

and that seems to resolve the error for that particular alert but it seems as though grafana is wanting me to do this to ALL of my alerts (as best as I can tell).  I'd really prefer to not have to do this since we have many alerts.

Here is an example of a complete alert, that was exported via the web UI and that caused the error above:

```
        - uid: 2xymZ3L4z
          title: DFT errors
          condition: C
          data:
            - refId: A
              relativeTimeRange:
                from: 600
                to: 0
              datasourceUid: prod2-cust-common-logs
              model:
                alias: ""
                bucketAggs:
                    - field: '@timestamp'
                      id: "2"
                      settings:
                        interval: auto
                      type: date_histogram
                datasource:
                    type: elasticsearch
                    uid: prod2-cust-common-logs
                hide: false
                intervalMs: 1000
                maxDataPoints: 43200
                metrics:
                    - id: "1"
                      type: count
                query: '"failed to apply delta to file"'
                refId: A
                timeField: '@timestamp'
            - refId: B
              relativeTimeRange:
                from: 600
                to: 0
              datasourceUid: __expr__
              model:
                conditions:
                    - evaluator:
                        params: []
                        type: gt
                      operator:
                        type: and
                      query:
                        params:
                            - B
                      reducer:
                        params: []
                        type: last
                      type: query
                datasource:
                    type: __expr__
                    uid: __expr__
                expression: A
                hide: false
                intervalMs: 1000
                maxDataPoints: 43200
                reducer: last
                refId: B
                type: reduce
            - refId: C
              relativeTimeRange:
                from: 600
                to: 0
              datasourceUid: __expr__
              model:
                conditions:
                    - evaluator:
                        params:
                            - 1
                        type: gt
                      operator:
                        type: and
                      query:
                        params:
                            - C
                      reducer:
                        params: []
                        type: last
                      type: query
                datasource:
                    type: __expr__
                    uid: __expr__
                expression: B
                hide: false
                intervalMs: 1000
                maxDataPoints: 43200
                refId: C
                type: threshold
          dashboardUid: ""
          panelId: 0
          noDataState: OK
          execErrState: OK
          annotations:
            summary: failed to apply delta to file error found
          labels:
            BMAP_PagerDuty: "True"
            Teams: "True"
          isPaused: false
```

and the modified version (just one line added) to get it to stop complaining about this alert but then it just moves onto the next one:

```
        - uid: 2xymZ3L4z
          title: DFT errors
          condition: C
          data:
            - refId: A
              relativeTimeRange:
                from: 600
                to: 0
              datasourceUid: prod2-cust-common-logs
              model:
                alias: ""
                bucketAggs:
                    - field: '@timestamp'
                      id: "2"
                      settings:
                        interval: auto
                      type: date_histogram
                datasource:
                    type: elasticsearch
                    uid: prod2-cust-common-logs
                hide: false
                intervalMs: 1000
                maxDataPoints: 43200
                metrics:
                    - id: "1"
                      type: count
                query: '"failed to apply delta to file"'
                refId: A
                timeField: '@timestamp'
            - refId: B
              relativeTimeRange:
                from: 600
                to: 0
              datasourceUid: __expr__
              model:
                conditions:
                    - evaluator:
                        params: []
                        type: gt
                      operator:
                        type: and
                      query:
                        params:
                            - B
                      reducer:
                        params: []
                        type: last
                      type: query
                datasource:
                    type: __expr__
                    uid: __expr__
                expression: A
                hide: false
                intervalMs: 1000
                maxDataPoints: 43200
                reducer: last
                refId: B
                type: reduce
            - refId: C
              relativeTimeRange:
                from: 600
                to: 0
              datasourceUid: __expr__
              model:
                conditions:
                    - evaluator:
                        params:
                            - 1
                        type: gt
                      operator:
                        type: and
                      query:
                        params:
                            - C
                      reducer:
                        params: []
                        type: last
                      type: query
                datasource:
                    type: __expr__
                    uid: __expr__
                expression: B
                hide: false
                intervalMs: 1000
                maxDataPoints: 43200
                refId: C
                type: threshold
          dashboardUid: ""
          panelId: 0
          noDataState: OK
          execErrState: OK
          for: 0s
          annotations:
            summary: failed to apply delta to file error found
          labels:
            BMAP_PagerDuty: "True"
            Teams: "True"
          isPaused: false
```

This is pretty odd to me that it just started happening.  We did hit this one other time a couple of weeks ago for the first time but it was only complaining about a single alert, we added the duration statement to that alert and grafana happily started up after that.  I'm not sure why it's now doing this for all of these other alerts.  We have not changed grafana versions recently.

### What did you expect to happen?

grafana to startup without errors with the yaml that it exported

### Did this work before?

yes, this has been working for us for quite some time

### How do we reproduce it?

1. see alert above that is causing issues
2.
3.

### Is the bug inside a dashboard panel?

no

### Environment (with versions)?

Grafana: 11.3.0
OS: kubernetes
Browser: chrome

### Grafana platform?

Kubernetes

**Comment by danfinn** [comment]
this sounds potentially related:
https://github.com/grafana/terraform-provider-grafana/issues/1273

**Comment by rwwiv** [comment]
Hi @danfinn, just to be sure this has started happening without any changes to the version of Grafana?

**Comment by rwwiv** [comment]
Looks like this was actually fixed with this PR: https://github.com/grafana/grafana/pull/94646

It should be in the next minor build of 11.4

**Comment by danfinn** [comment]
@rwwiv I can't be totally sure because we did see this one other time as I mentioned a couple of weeks ago but didn't think a whole lot of it because it was only complaining about a single alert and once we added the `for: 0s` to that alert grafana started up without any errors.  We did however upgrade to 11.3.0 sometime around that time that we had the one off alert cause an error but unfortunately I don't have the exact dates to say for sure but it could definitely be something that was caused by upgrading to 11.3.

When is that 11.4 docker image coming out?  This is causing us major issues in production, we cannot make changes or restart our existing grafana pods because of this.

**Comment by danfinn** [comment]
also @rwwiv can you clarify which version of 11.4 this will be in?  11.4.1?

**Comment by rwwiv** [comment]
Sorry @danfinn, that was a typo on my end. Next minor build of Grafana 11, which based on our release cadence should be coming in January but doesn't have a set release date yet.

**Comment by danfinn** [comment]
oh, that is not good at all for us.  What are my options for resolving this now?  Do I need to add a `for: 0s` to every single alert?

**Comment by rwwiv** [comment]
Clicked the wrong button and deleted my comment, oops. Thanks for capturing that!

---

> Do I need to add a for: 0s to every single alert?

It sounds like this is only a problem for alerts you're exporting from the UI, is that correct? Unfortunately until that fix makes it into a release yes you will have to add that field to the exported rule(s) before committing them to your git repo.

**Comment by danfinn** [comment]
because we are using the gitops model we export ALL of our alerts.  This is quite annoying as we have hundreds of them.

**Comment by danfinn** [comment]
I should have looked closer, I mistakingly assumed all of our alerts needed this but it seems only alerts that have been added recently were missing this duration statement.  It seems like in the past grafana used to include this with exported alerts and then at some point recently stopped.

**Comment by vi-simon** [comment]
same issue, we use the same workflow - create alert in the UI, export it - provision.  - exporting yaml alert:
```apiVersion: 1
    groups:
        - orgId: 1
          name: evaluate-1m
          folder: backend-logs-services-alerts
          interval: 1m
          rules:
            - uid: cedgf35r9o0zka
              title: Backend-services-alerts
              condition: Threshold
              data:
                - refId: Query
                  queryType: range
                  relativeTimeRange:
                    from: 3600
                    to: 0
                  datasourceUid: P8E80F9AEF21F6940
                  model:
                    datasource:
                        type: loki
                        uid: P8E80F9AEF21F6940
                    editorMode: code
                    expr: |-
                        ###my expression
                    intervalMs: 60000
                    legendFormat: ""
                    maxDataPoints: 43200
                    queryType: range
                    refId: Query
                - refId: Reduce
                  datasourceUid: __expr__
                  model:
                    conditions:
                        - evaluator:
                            params: []
                            type: gt
                          operator:
                            type: and
                          query:
                            params:
                                - B
                          reducer:
                            params: []
                            type: last
                          type: query
                    datasource:
                        type: __expr__
                        uid: __expr__
                    expression: Query
                    intervalMs: 1000
                    maxDataPoints: 43200
                    reducer: last
                    refId: Reduce
                    type: reduce
                - refId: Threshold
                  datasourceUid: __expr__
                  model:
                    conditions:
                        - evaluator:
                            params:
                                - 0
                            type: gt
                          operator:
                            type: and
                          query:
                            params:
                                - C
                          reducer:
                            params: []
                            type: last
                          type: query
                    datasource:
                        type: __expr__
                        uid: __expr__
                    expression: Reduce
                    intervalMs: 1000
                    maxDataPoints: 43200
                    refId: Threshold
                    type: threshold
              dashboardUid: dedcemvcqx7uoe
              panelId: 1
              noDataState: OK
              execErrState: Error
              for: 0s
              annotations:
                __dashboardUid__: dedcemvcqx7uoe
                __panelId__: "1"
              isPaused: false
              notification_settings:
                receiver: slack
```

need to add ```for: 0s``` to solve this 
kube-prometheus-stack helm version chart 66.3.0
grafana version  v11.3.1 (64b556c137)

**Comment by rwwiv** [comment]
Closing as fixed by https://github.com/grafana/grafana/pull/94646

Please upgrade to a version of Grafana that includes the fix and feel free to open a new issue if the problem persists after that.

**Comment by danfinn** [comment]
@rwwiv can you confirm which versions of grafana will have this fix?  Would 11.5.2 have it?

**Comment by danfinn** [comment]
I just tested against 11.5.2 and it still seems to be doing the same thing, not including the duration statement on new alerts created in the UI.  Please confirm which version I should be using to test to see if this is fixed.  It looks like all the recent docker image versions on dockerhub were created prior to you saying this was fixed so I think I might be waiting on a new version to get pushed to dockerhub.
